### PR TITLE
RavenDB-11058 added support for attachment rename

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

**Review by fitzchak — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/BatchRequestParser.cs [hunk 1]
```diff
@@ -661,6 +681,11 @@ private static unsafe CommandPropertyName GetPropertyType(JsonParserState state)
                     if (*(int*)state.StringBuffer == 1836020294 &&
                         state.StringBuffer[4] == (byte)'E')
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/BatchRequestParser.cs]: This is also wrong. `FromETM` will be accepted here.

File: src/Raven.Server/Documents/Handlers/BatchRequestParser.cs [hunk 2]
```diff
@@ -661,6 +681,11 @@ private static unsafe CommandPropertyName GetPropertyType(JsonParserState state)
...
                         return CommandPropertyName.FromEtl;
+
+                    if (*(int*)state.StringBuffer == 1316447566 &&
+                        state.StringBuffer[3] == (byte)'N')
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/BatchRequestParser.cs]: This is wrong. You compare `NewN` only. 
This will accept `NewNumb`

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/AttachmentsStorage.cs [hunk 3]
```diff
@@ -761,6 +776,28 @@ private static void ThrowConcurrentExceptionOnMissingAttachment(string documentI
...
+        public AttachmentDetails RenameAttachment(DocumentsOperationContext context, string documentId, string name, string newName, LazyStringValue changeVector)
+        {
+            if (string.IsNullOrWhiteSpace(documentId))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(documentId));
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(name));
+            if (string.IsNullOrWhiteSpace(newName))
+                throw new ArgumentException("Argument cannot be null or whitespace.", nameof(newName));
+            if (context.Transaction == null)
+                throw new ArgumentException("Context must be set with a valid transaction before calling Get", nameof(context));
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/AttachmentsStorage.cs]: Minor: "... before calling Get" -> "... before calling rename"

**Review by arekpalinski — APPROVED** [review]

**Comment by ppekrol** [comment]
Fixed

**Pull request merged**